components/backup/src/endpoint.rs

```diff
@@ -174,6 +183,70 @@ impl BackupRange {
         let stat = scanner.take_statistics();
         Ok(stat)
     }
+
+    fn backup_raw<E: Engine>(
+        &self,
+        writer: &mut BackupRawKVWriter,
+        engine: &E,
+        backup_ts: TimeStamp,
+    ) -> Result<Statistics> {
+        let _ = backup_ts;
+        let mut ctx = Context::default();
+        ctx.set_region_id(self.region.get_id());
+        ctx.set_region_epoch(self.region.get_region_epoch().to_owned());
+        ctx.set_peer(self.leader.clone());
+        let snapshot = match engine.snapshot(&ctx) {
+            Ok(s) => s,
+            Err(e) => {
+                error!("backup snapshot failed"; "error" => ?e);
+                return Err(e.into());
+            }
+        };
+        let start = Instant::now();
+
+        let mut option = IterOption::default();
+        if let Some(end) = self.end_key.clone() {
+            option.set_upper_bound(end.as_encoded(), DATA_KEY_PREFIX_LEN);
+        }
+        let mut cursor = snapshot.iter_cf(rawkv_cf(&self.cf)?, option, ScanMode::Forward)?;
+        let raw: Vec<u8> = (0..0).collect();
+        let empty_key = Key::from_raw(&raw);
+        let s_key = match self.start_key.clone() {
+            Some(s) => s,
+            None => empty_key,
+        };
+        let mut statistics = Statistics::default();
+        let cfstatistics = statistics.mut_cf_statistics(&self.cf);
+        if !cursor.seek(&s_key, cfstatistics)? {
```

Comment:
Actually if s_key is not set, you can use `cursor.seek_to_first` instead of creating an empty key.